code/iaas/auth-logic/src/main/java/io/cattle/platform/iaas/api/auth/dao/impl/PasswordDaoImpl.java

```diff
@@ -51,46 +54,65 @@ public Credential changePassword(Credential password, ChangePassword changePassw
-    public Credential updateAdminAndCredentials(String username, String password, String name) {
-        Account admin = create()
-                .selectFrom(ACCOUNT)
-                .where(ACCOUNT.STATE.eq(CommonStatesConstants.ACTIVE)
-                        .and(ACCOUNT.KIND.eq(AccountConstants.ADMIN_KIND)))
-                .orderBy(ACCOUNT.ID.asc()).limit(1).fetchOne();
-
-        if (admin == null){
-            throw new ClientVisibleException(ResponseCodes.INTERNAL_SERVER_ERROR, "UnableToFindAdmin");
+    public Credential createAdminAccountAndPassword(String username, String password, String name) {
+        if (StringUtils.isBlank(username) || StringUtils.isBlank(password)) {
+            throw new ClientVisibleException(ResponseCodes.BAD_REQUEST, "UserCredentialCreation", "Cannot have blank username or password.", null);
+        }
+        Account account;
+        Credential credential = create()
+                .selectFrom(CREDENTIAL)
+                .where(CREDENTIAL.STATE.eq(CommonStatesConstants.ACTIVE))
+                .and(CREDENTIAL.PUBLIC_VALUE.eq(username)
+                .and(CREDENTIAL.KIND.equalIgnoreCase(CredentialConstants.KIND_PASSWORD)))
+                .fetchOne();
+
+        SelectConditionStep<CredentialRecord> query =
+                create().selectFrom(CREDENTIAL)
+                        .where(CREDENTIAL.PUBLIC_VALUE.eq(username)
+                                .and(CREDENTIAL.STATE.eq(CommonStatesConstants.ACTIVE)
+                                        .and(CREDENTIAL.KIND.eq(CredentialConstants.KIND_PASSWORD))));
+        if (credential != null ) {
+            query = query.and(CREDENTIAL.ACCOUNT_ID.ne(credential.getAccountId()));
         }
 
-         if (create().selectFrom(CREDENTIAL)
-                .where(CREDENTIAL.PUBLIC_VALUE.eq(username)
-                        .and(CREDENTIAL.STATE.eq(CommonStatesConstants.ACTIVE)
-                                .and(CREDENTIAL.KIND.eq(CredentialConstants.KIND_PASSWORD)
-                                .and(CREDENTIAL.ACCOUNT_ID.ne(admin.getId())))))
-                .fetchAny() != null){
-             throw new ClientVisibleException(ResponseCodes.CONFLICT, "UsernameIsTaken", "Username: " + username +" is taken", null);
-         }
-
-        create().delete(CREDENTIAL)
-                .where(CREDENTIAL.ACCOUNT_ID.eq(admin.getId())
-                        .and(CREDENTIAL.KIND.eq(CredentialConstants.KIND_PASSWORD)))
-                .execute();
+        if (query.fetchAny() != null){
+            throw new ClientVisibleException(ResponseCodes.CONFLICT, "UsernameIsTaken", "Username: " + username +" is taken", null);
+        }
 
         Map<Object, Object> properties = new HashMap<>();
 
-        if (StringUtils.isBlank(username) || StringUtils.isBlank(password)) {
-            throw new ClientVisibleException(ResponseCodes.BAD_REQUEST, "UserCredentialCreation", "Cannot have blank username or password.", null);
+        if (credential != null) {
```

Comment:
Basically. Was originally a different query. Need to rework this entire method, been thinking about how to do this all day. I think I almost have it.